services/app-api/resolvers/submitDraftSubmission.ts

```diff
@@ -148,8 +162,47 @@ export function submitDraftSubmissionResolver(
             )
         }
 
+        const draftResult = toDomain(currentRevision.submissionFormProto)
+
+        if (draftResult instanceof Error) {
+            const errMessage = `Failed to decode draft proto ${draftResult}.`
+            logError('submitDraftSubmission', errMessage)
+            throw new Error(errMessage)
+        }
+
+        if (draftResult.status === 'SUBMITTED') {
+            const errMessage = `Attempted to submit and already submitted package.`
+            logError('submitDraftSubmission', errMessage)
+            throw new Error(errMessage)
+        }
+
+        //Set submitInfo default to initial submission
+        const submitInfo: UpdateInfoType = {
+            updatedAt: new Date(),
+            updatedBy: context.user.email,
+            updatedReason: 'Initial submission'
+        }
+
+        //If submission is a resubmission set submitInfo updated reason to input.
+        if (planPackageStatus === 'UNLOCKED' && submittedReason ) {
+            submitInfo.updatedReason = submittedReason
```

Comment:
@haworku This looks good to me, setting the default value is happening at submit time and handling either setting the default in the initial submission case, or setting the value passed in on re-submission is handled all by the resolver code.

Reply:
Ohhhhh. Thank you for calling it out I was looking for this code. 🔍